File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 14/07/2026, 14:31:42
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n=len(prices)
        min=prices[0]
        max_profit=0
        for num in prices:
            if(num<min):
                min=num
            elif(num - min > max_profit):
                max_profit=num-min
        return max_profit

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 14/07/2026, 14:31:47
class Solution 
{
    public int maxProfit(int[] arr) 
    {
        int buy=arr[0];
        int max=0;
        // int min=Integer.MAX_VALUE;
        
        // for(int i=0;i<n;i++)
        // {
        //     if(arr[i]<min)
        //     {
        //         min=arr[i];
        //     }
        //     else if(arr[i]-min>max)
        //     {
        //         max=arr[i]-min;
        //     }
        // }
        // return max;
        for(int i=1;i<arr.length;i++)
        {
            if(buy>arr[i])
            {
                buy=arr[i];
            }
            else
            {
                int p=arr[i]-buy;
                if(max<p)
                {
                    max=p;
                }
            }
        }
return max;
    }
}